integreat_cms/deepl_api/utils.py

```diff
@@ -0,0 +1,161 @@
+import logging
+import deepl
+
+from django.conf import settings
+from django.contrib import messages
+from django.utils.translation import ugettext as _
+
+logger = logging.getLogger(__name__)
+
+
+class DeepLApi:
+    """
+    DeepL API to auto translate selected posts.
+    """
+
+    def __init__(self):
+        """
+        Initialize the DeepL client
+        """
+        self.translator = deepl.Translator(settings.DEEPL_AUTH_KEY)
+
+    def check_availability(self, request, language_slug):
+        """
+        This function checks, if the selected language is supported by DeepL
+
+        :param request: request that was sent to the server
+        :type request: ~django.http.HttpRequest
+
+        :param language_slug: current language slug
+        :type language_slug: str
+
+        :return: true or false
+        :rtype: bool
+        """
+        supported_source_languages = [
+            source_language.code.lower()
+            for source_language in self.translator.get_source_languages()
+        ]
+        supported_target_languages = [
+            target_languages.code.lower()[:2]
+            for target_languages in self.translator.get_target_languages()
+        ]
+        return (
+            request.region.default_language.slug in supported_source_languages
+            and language_slug in supported_target_languages
+        )
+
+    def get_translation(self, content, source_language_slug, target_language_key):
+        """
+        This functions gets the translation from DeepL
+
+        :param content: passed content
+        :type content: str
+
+        :param source_language_slug: slug of source language
+        :type source_language_slug: str
+
+        :param target_language_key: slug or in case of pt and en bcp of target language
+        :type target_language_key: str
+
+        :return: translated post in target language
+        :rtype: str
+        """
+        translated_content = self.translator.translate_text(
+            content, source_lang=source_language_slug, target_lang=target_language_key
+        )
+        return translated_content
+
+    def deepl_translation(
+        self, request, content_translations, language_slug, form_class, content_label
```

Comment:
```suggestion
        self, request, content_objects, language_slug, form_class
```
The `content_label` can be derived from the translation objects and `content_translations` isn't an ideal variable name because it contains the content objects, not the translation objects.